src/support/error_handling.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file err_handling.h
+ * \brief Common error handling functions for socket.h and pipe.h
+ */
+#ifndef TVM_SUPPORT_ERR_HANDLING_H_
+#define TVM_SUPPORT_ERR_HANDLING_H_
+#include <errno.h>
+
+namespace tvm {
+namespace support {
+   /*!
+   * \return last error of socket operation
+   */
+  static int GetLastErrorCode() {
```

Comment:
This needs to be passed as a second template function parameter to RetryCallOnEINTR

Reply:
We need specialized function in socket and pipe

For socket it should be `WSAGetLastError()` in windows, for pipe it should be `GetLastError()`